httpcore/_backends/anyio.py

```diff
@@ -167,11 +170,13 @@ async def open_uds_stream(
     ) -> AsyncSocketStream:
         connect_timeout = timeout.get("connect")
         unicode_host = hostname.decode("utf-8")
-        exc_map = {
-            OSError: ConnectError,
-            TimeoutError: ConnectTimeout,
-            BrokenResourceError: ConnectError,
-        }
+        exc_map = OrderedDict(
+            [
+                (TimeoutError, ConnectTimeout),
+                (OSError, ConnectError),
+                (BrokenResourceError, ConnectError),
+            ]
+        )
```

Comment:
Dicts are officially ordered since 3.7, and have been in CPython as an implementation detail since 3.5.

So… Could we do without `OrderedDict`, at least _for now_*?

```suggestion
        exc_map = {
            TimeoutError: ConnectTimeout,
            OSError: ConnectError,
            BrokenResourceError: ConnectError,
        }
```

(*I understand this wouldn't be guaranteed to work on eg Pypy, and this may not convey the idea that "order matters" particularly well. There's a case to be made to perhaps switch `exc_map` to accept a _sequence_ of 2-tuples, but we'd need to treat it as a follow-up to keep this PR focused on fixing the bug, rather than any extra refactoring.)

Reply:
Yes, I'd get rid of `OrderedDict`.

Anyway I throw away evetyhing not related with the `asyncio` changes